lib/blob/blobstore.c

```diff
@@ -7461,6 +7489,9 @@ delete_snapshot_sync_clone_cpl(void *cb_arg, int bserrno)
 	/* Clear cluster map entries for snapshot */
 	for (i = 0; i < ctx->snapshot->active.num_clusters && i < ctx->clone->active.num_clusters; i++) {
 		if (ctx->clone->active.clusters[i] == ctx->snapshot->active.clusters[i]) {
+			if (ctx->snapshot->active.clusters[i] != 0) {
+				ctx->snapshot->active.num_allocated_clusters--;
```

Comment:
Same here.

Reply:
Every time an element of the cluster map `active.clusters` change from 0 to a different value or from a value != 0 to 0, we must update `active.num_allocated_clusters` as well.
In this case an element is setted to 0, so if before it was != 0 we are decreasing of 1 cluster.